docs-src/tls.md

```diff
@@ -0,0 +1,138 @@
+# TLS
+
+Service APIs allow for a variety of ways to configure TLS. This document lays
+out various TLS settings and gives general guidelines on how to use them
+effectively.
+
+## Client/Server and TLS
+
+![overview](tls-overview.svg)
+
+For Gateways, there are two connections involved:
+
+- **downstream**: This is the connection between the client and the Gateway.
+- **upstream**: This is the connection between the Gateway and backend resources
+   specified by routes. These backend resources will usually be Services.
+  
+With Service APIs, TLS configuration of downstream and
+upstream connections is managed independently.
+
+Please note that in case of `Passthrough` TLS mode, no TLS settings take
+effect as the TLS session from the client is NOT terminated at the Gateway.
+The rest of the document assumes that TLS is being terminated at the Gateway.
```

Comment:
Perhaps state that "Terminate" is the default.  I noticed that some examples specify `mode: Terminate` and some omit it (but get `Terminate` because it is the default).